lib/strings.nix

```
@@ -238,7 +243,12 @@ rec {
     # Prefix to check for
     pref:
     # Input string
-    str: substring 0 (stringLength pref) str == pref;
+    str:
+    # pref must be a string
+    # str must be a string-like
+    if isPath pref then throw ''
+      lib.strings.hasPrefix: The first argument (${toString pref}) is a path, which is not supported''
+    else (substring 0 (stringLength pref) str == pref);
```

```suggestion
    else
      substring 0 (stringLength pref) str == pref;
```

Nothing fits on screen anyway because of the large doc strings. Might as well improve readability.